src/link_prober/LinkProber.cpp

```diff
@@ -313,25 +338,41 @@ void LinkProber::handleRecv(
                 mMuxPortConfig.getPortName() %
                 mMuxPortConfig.getBladeIpv4Address().to_string()
             );
-            if (memcmp(icmpPayload->un.uuid.data, IcmpPayload::getGuidData(),
-                       IcmpPayload::getGuid().size()) == 0) {
+            bool isMatch = (memcmp(icmpPayload->uuid, IcmpPayload::getGuidData(), sizeof(icmpPayload->uuid)) == 0);
+            if (isMatch)
+            {
                 MUXLOGTRACE(boost::format("%s: Matching Guid") % mMuxPortConfig.getPortName());
                 // echo reply for an echo request generated by this/active ToR
                 mRxSelfSeqNo = mTxSeqNo;
                 mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpSelfEvent());
-            } else {
-                // echo reply for an echo request generated by peer ToR
-                mRxPeerSeqNo = mTxSeqNo;
-                mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpPeerEvent());
-                if (ntohl(icmpPayload->command) == static_cast<uint32_t> (Command::COMMAND_SWITCH_ACTIVE)) {
-                    boost::asio::io_service::strand &strand = mLinkProberStateMachine.getStrand();
-                    boost::asio::io_service &ioService = strand.context();
-                    ioService.post(strand.wrap(boost::bind(
-                        static_cast<void (LinkProberStateMachine::*) (SwitchActiveRequestEvent&)>
-                            (&LinkProberStateMachine::processEvent),
-                        &mLinkProberStateMachine,
-                        LinkProberStateMachine::getSwitchActiveRequestEvent()
-                    )));
+            }
+
+            uint8_t tlvStartOffset = mPacketHeaderSize + sizeof(IcmpPayload);
+            bool stopProcessTlv = false;
+            while (tlvStartOffset + sizeof(Tlv) <= bytesTransferred) {
```

Comment:
Suggest adding member function of append_tlv and get_next_tlv to encapsulate the operations and add unit test to test these 2 methods generically.

Reply:
added methods:
* `findNextTlv`: find the next TLV from the offset
* `appendTlvCommand`: append TlvCommand to the end of tx buffer
* `appendTlvSentinel`: append TlvSentinel to the end of tx buffer